### PR TITLE
fix(components): [dropdown] add disabled support

### LINKED ISSUE #6366: [Bug Report] [Component] [dropdown] el-dropdown属性disabled不能生效的问题
Bug Type: **`Component`**

## Environment

- Vue Version: `3.2.31`
- Element Plus Version: `2.0.4`
- Browser / OS: `Chrom98.0.4758.102（正式版本）/ windows 10 家庭中文版`
- Build Tool: `Vue CLI`

### Related Component

- `el-dropdown`

### Reproduction Link

[Element Plus Playground](https://element-plus.run/#eyJBcHAudnVlIjoiPHRlbXBsYXRlPlxuICA8IS0tIOW9k2Rpc2FibGVk5Y+W5YC85Li6dHJ1ZeaXtu+8jGVsLWRyb3Bkb3du5bm25pyq56aB55SoIC0tPlxuICA8ZWwtZHJvcGRvd24gOmRpc2FibGVkPVwidHJ1ZVwiIEBjb21tYW5kPVwib25EZWxpdmVyeVwiPlxuICAgIDxlbC1idXR0b24gdHlwZT1cInByaW1hcnlcIiBzaXplPVwic21hbGxcIiA6ZGlzYWJsZWQ9XCJ0cnVlXCI+XG4gICAgICDlh7rlupM8ZWwtaWNvbj48QXJyb3dEb3duIC8+PC9lbC1pY29uPlxuICAgIDwvZWwtYnV0dG9uPlxuICAgIDx0ZW1wbGF0ZSAjZHJvcGRvd24+XG4gICAgICA8ZWwtZHJvcGRvd24tbWVudT5cbiAgICAgICAgPGVsLWRyb3Bkb3duLWl0ZW0gOmNvbW1hbmQ9XCIxXCI+5oOF5Ya1MTwvZWwtZHJvcGRvd24taXRlbT5cbiAgICAgICAgPGVsLWRyb3Bkb3duLWl0ZW0gOmNvbW1hbmQ9XCIzXCI+5oOF5Ya1MjwvZWwtZHJvcGRvd24taXRlbT5cbiAgICAgICAgPGVsLWRyb3Bkb3duLWl0ZW0gOmNvbW1hbmQ9XCIzXCI+5oOF5Ya1MzwvZWwtZHJvcGRvd24taXRlbT5cbiAgICAgIDwvZWwtZHJvcGRvd24tbWVudT5cbiAgICA8L3RlbXBsYXRlPlxuICA8L2VsLWRyb3Bkb3duPlxuPC90ZW1wbGF0ZT5cblxuPHNjcmlwdCBzZXR1cCBsYW5nPVwidHNcIj5cbmltcG9ydCB7IEFycm93RG93biB9IGZyb20gJ0BlbGVtZW50LXBsdXMvaWNvbnMtdnVlJ1xuXG5jb25zdCBvbkRlbGl2ZXJ5OnZvaWQgPSAodmFsKSA9PiB7XG4gIFx0Y29uc29sZS5sb2codmFsKVxufVxuPC9zY3JpcHQ+IiwiaW1wb3J0LW1hcC5qc29uIjoie1wiaW1wb3J0c1wiOnt9fSJ9)

### Steps to reproduce

```vue
<template>
  <!-- 当disabled取值为true时，el-dropdown并未禁用 -->
  <el-dropdown :disabled="true" @command="onDelivery">
    <el-button type="primary" size="small" :disabled="true">
      出库<el-icon><ArrowDown /></el-icon>
    </el-button>
    <template #dropdown>
      <el-dropdown-menu>
        <el-dropdown-item :command="1">情况1</el-dropdown-item>
        <el-dropdown-item :command="3">情况2</el-dropdown-item>
        <el-dropdown-item :command="3">情况3</el-dropdown-item>
      </el-dropdown-menu>
    </template>
  </el-dropdown>
</template>

<script setup lang="ts">
import { ArrowDown } from '@element-plus/icons-vue'

const onDelivery:void = (val) => {
  	console.log(val)
}
</script>
```

### What is Expected?

当设置el-dropdown的disabled属性为true时，禁用hover行为，下拉菜单不展示

### What is actually happening?

下拉菜单仍然展示

### Additional comments

_(empty)_

### PR DESCRIPTION
**Pull request by freedomlang**
Please make sure these boxes are checked before submitting your PR, thank you!

- [x] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [x] Make sure you are merging your commits to `dev` branch.
- [x] Add some descriptions and refer to relative issues for your PR.

fix #6366

### CODE REVIEW

**Comment by Felix-lx** [comment]
this fix will be released later in the elementPlus version？

File: packages/components/dropdown/src/dropdown.vue [hunk 1]
```diff
@@ -1,5 +1,5 @@
 <template>
-  <div :class="ns.b()">
+  <div :class="[ns.b(), { disabled }]">
```
Comment (msidolphin) [inline packages/components/dropdown/src/dropdown.vue:2]: please use `is-disabled` instead of disabled

Comment (freedomlang) [reply packages/components/dropdown/src/dropdown.vue]: Resolved

ps: I just search the whole project, and found `disabled` classname in other files. I think they should use `is-disabled` too. If it does, i'll start another pr to do it

**Comment by JeremyWuuuuu** [comment]
> this fix will be released later in the elementPlus version？

@Felix-lx  yes it will

File: packages/components/dropdown/src/dropdown.vue (hunk 1 above)
Comment (JeremyWuuuuu) [reply packages/components/dropdown/src/dropdown.vue]: @freedomlang Sure! Let's refactor them!

Comment (JeremyWuuuuu) [reply packages/components/dropdown/src/dropdown.vue]: @freedomlang seems like you didn't do `ns.is('disabled', disabled)` as @msidolphin asked. Would you mind update it? I am preparing a release now

Comment (freedomlang) [reply packages/components/dropdown/src/dropdown.vue]: > @freedomlang seems like you didn't do `ns.is('disabled', disabled)` as @msidolphin asked. Would you mind update it? I am preparing a release now

I'll check it in a secsond

Comment (freedomlang) [reply packages/components/dropdown/src/dropdown.vue]: > > @freedomlang seems like you didn't do `ns.is('disabled', disabled)` as @msidolphin asked. Would you mind update it? I am preparing a release now
> 
> I'll check it in a secsond

Done

**Review by jw-foss — APPROVED** [review]

**Pull request merged**